guidelines/monai-application-package.md

```diff
@@ -1,498 +1,588 @@
 
-    - Example legal values: `1`, `42`, `2048`
+      - Example: `1.5Gi`, `2048Mi`
 
-    - Example illegal values: `-1`, `1.5`, `2,048`
+  - GPU Memory requirements SHALL be denoted using decimal values followed by units (`/etc/holoscan/pkg.json#resources.gpu-memory`).
 
-  - Decimal values MUST be positive, rounded to nearest tenth, use the `.` character to separate whole and fractional values, and not contain any positional separators.
+    - Supported units SHALL be mebibytes (`Mi`) and gibibytes (`Gi`).
 
-    - Example legal values: `1`, `1.0`, `0.5`, `2.5`, `1024`
+      - Example: `1.5Gi`, `2048Mi`
 
-    - Example illegal values: `1,024`, `-1.0`, `3.14`
-
-  - When not provided the default values of `cpu=1`, `gpu=0`, `memory="1Gi"`, and `shared-memory="64Mi"` will be assumed.
+  - Shared memory requirements SHALL be denoted using decimal values followed by units (`/etc/holoscan/pkg.json#resources.shared-memory`).
 
+    - Supported units SHALL be mebibytes (`Mi`) and gibibytes (`Gi`).
 
-#### Table of Package Manifest Fields
+      - Example: `1.5Gi`, `2048Mi`
 
-| Name                     | Required | Default | Type    | Format                    | Description                                                                  |
-| ------------------------ | -------- | ------- | ------- | ------------------------- | ---------------------------------------------------------------------------- |
-| `api-version`            | No       | 0.0.0   | string  | semantic version          | Version of the manifest file schema.                                         |
-| `application-root`       | **Yes**  | N/A     | string  | absolute file-system path | Absolute file-system path to the folder older which contains the Application |
-| `models`                 | No       | N/A     | array   | array of objects          | Array of objects which describe models in the package.                       |
-| `models[*].name`         | No       | N/A     | string  | map reference name        | Name of the map which conforms to the map naming rules.                      |
-| `models[*].path`         | **Yes**  | N/A     | string  | absolute file-system path | Absolute file-system path to the folder which contains the model.            |
-| `resources`              | No       | N/A     | object  | object                    | Object describing resource requirements for the Application.                 |
-| `resources.cpu`          | No       | 1       | integer | number                    | Number of CPU cores required by the Application.                             |
-| `resources.gpu`          | No       | 0       | integer | number                    | Number of GPU devices required by the Application.                           |
-| `resource.memory`        | No       | 1Gi     | string  | memory size               | The maximum memory required by the Application.                              |
```

Comment:
Should we define a minimum and maximum? It may be that a GPU is not essential and it may be that a code will only run on a single GPU.
Also what about min GPU memory requirements?

Reply:
These are all minimums at the moment. 
I am working with @gigony to see where we need the "maximum" defined.

Reply:
@JHancox We have added resource `limits` to all items; please review the latest commit and let us know.

Reply:
Looks comments are on the deleted line (in the existing spec doc), but the changed ones are similar with additions of gpu.memory and fragment specific sections.

Anyway, based on what it is declared the said values are minimums? They are merely Defaults. Please be reminded, the values are supposed to come from the app developers, and who is to say they cannot state a memory usage of 500MiB as the CAP/MAX of system memory usage by the app. The minimum system mem allocated for a MAP container at runtime, at least over and above the stated value in the MAP, is up to the hosting app/platform, a separate concern.

Again, the MAX values come from the app, not the SDK in theory, and should be below the theoretical limit, though in practical terms, much lower, 1 TB?